### DISCUSSION TITLE
CondSRF : does it retain anistropy from variogram model in Krige class?

### DISCUSSION THREAD

**Discussion by wtrainor** [GeoStat-Framework/GSTools#208] [Q&A]
Hello,

thanks for your package!

I've been looking for a python version of sequential guassian simulation, and thought maybe your conditioned random field functionality does something similar: stochastic results that reproduce the variogram and histogram of the conditioned data. But it appears from the documentation, that you can only have isotropic conditioned random fields in GStools? Is that true?

many thanks,
whitney

**Comment by MuellerSeb** [comment]
Hey there!

It **is possible** to have anisotrope conditioned random fields! 

Seems like we have to be more clear about that in the documentation.
Do you remember the line in the docs that implied that? Maybe because we don't have any estimation routines for anisotropy yet?
Sorry for the inconvenience!

Here an example:
```python
import numpy as np
import gstools as gs

# conditioning data
cond_x = [0.3, 1.9, 1.1, 3.3, 4.7]
cond_y = [1.2, 0.6, 3.2, 4.4, 3.8]
cond_pos = [cond_x, cond_y]
cond_val = [0.47, 0.56, 0.74, 1.47, 1.74]

# grid definition for output field
x = np.arange(0.0, 5.5, 0.1)
y = np.arange(0.0, 6.5, 0.1)

# anisotrope and rotated model
model = gs.Gaussian(
    dim=2, len_scale=1, anis=0.2, angles=-0.5, var=0.5, nugget=0.1
)
krige = gs.Krige(model, cond_pos=cond_pos, cond_val=cond_val)
cond_srf = gs.CondSRF(krige)
cond_srf.structured([x, y])

cond_srf.plot()
cond_srf.krige.plot("krige_var")
```
Field:
![Figure_1](https://user-images.githubusercontent.com/19690642/132952155-c6ce0078-7a44-4a19-99ae-6b803237c37f.png)

Error-variance:
![Figure_2](https://user-images.githubusercontent.com/19690642/132952167-446b5728-f7cd-45fa-9c26-336d6b497493.png)

Cheers,
Sebastian

**Reply by wtrainor** [reply]
Thanks so much for the speedy reply @MuellerSeb.  So what I read was on page 358 of [the manual](https://readthedocs.org/projects/gstools/downloads/pdf/latest/).
"Randomization method for calculating isotropic random fields."
And I didn't see an alternative for `generator` that seemed more appropriate
Specifically, I'd like to confirm if these conditioned random methods are designed to reproduce the variogram.  Roughly my order of operations (after confirming that my angle of highest correlation is 45 degrees/0.785 radians):
```python
bin_center, gamma_45 = gs.vario_estimate(GSV_cond_pos, field,angles=0.78539816) 
GSVfit_model_Exp =gs.Exponential(dim=2,,angles=[0.78539816], anis=0.25)
GSVfit_model_Exp.fit_variogram(bin_center, gamma_45, nugget=True)
krigeOK_Exp = gs.krige.Ordinary(GSVfit_model_Exp, cond_pos=GSV_cond_pos, cond_val=GSV_cond_val, 
    normalizer=normalizer, fit_normalizer=myFitNormalizer,exact=True,fit_variogram=True)
krigeOK_Exp.structured((gridx, gridy))
cond_srf = gs.CondSRF(krigeOK_Exp)
ens_no = 4
field = []
for i in range(ens_no):
    field.append(cond_srf.structured([gridx, gridy],seed=i))
```
And the resulting 4 realizations don't seem to retain the same anistropy or even have the same variogram.
![image](https://user-images.githubusercontent.com/20952996/133406957-19843475-ee8b-4be0-93cd-43d412fa9bf7.png)

Your example is promising, so I suppose I should get rid of the `fit_variogram` step?

**Reply by MuellerSeb** [reply]
Sorry for the late reply. This looks strange.

First, since you are fitting your model in the third line, you shouldn't set `fit_variogram=True`.

Unfortunately, I couldn't reproduce this behavior. What are the estimated parameters? (`print(GSVfit_model_Exp)` after calling `fit_variogram(..)`)
